src/navigation/path.cpp

```diff
@@ -139,12 +148,49 @@ std::vector<glm::dvec3> Path::controlPoints() const {
 CameraPose Path::traversePath(double dt, float speedScale) {
     double speed = speedAlongPath(_traveledDistance);
     speed *= static_cast<double>(speedScale);
-    const double displacement =  dt * speed;
+    double displacement =  dt * speed;
+
+    const double prevDistance = _traveledDistance;
 
     _progressedTime += dt;
     _traveledDistance += displacement;
 
-    return interpolatedPose(_traveledDistance);
+    CameraPose newPose;
+
+    if (_type == Type::Linear) {
+        // Special handling of linear paths, so that it can be used when we are
+        // traversing very large distances without introducing precision problems
+        const glm::dvec3 prevPosToEnd = _prevPose.position - _end.position();
+        const double remainingDistance = glm::length(prevPosToEnd);
+
+        // Actual displacement may not be bigger than remaining distance
+        if (displacement > remainingDistance) {
+            displacement = remainingDistance;
+            _traveledDistance = pathLength();
+            _forceQuit = true;
```

Comment:
Does this mean that if _forceQuit is set to true, there would be one more tick of progressing along the path?  I'd intuitively assume that it would stop immediately, meaning that we would be `displacement-remainingDistance` away from the the target location when the path finishes?

Reply:
No, this is the final tick along the path and should result in the target position being returned (see the next line).
The check for whether a path has finished is done as the last step in the updateCamera function of `PathNavigator`. 

I get where you are coming from though. "Force" sounds very serious and immediate. Maybe just remaining the variable to `_shouldQuit` would be sufficient?

Note: The reason the variable is needed is to handle the cases where precision might be a problem. When the distance es are large, the check `(_traveledDistance / pathLength()) >= 1.0` is not sufficient to identify if a path has finished

Reply:
:+1:  `_shouldQuit` would make that more apparent imho.  But also not a big deal